pkg/lockfile/parse-apk-installed.go

```diff
@@ -0,0 +1,91 @@
+package lockfile
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+const AlpineEcosystem Ecosystem = "Alpine"
+
+func ParseApkInstalled(pathToLockfile string) ([]PackageDetails, error) {
+	var packages []PackageDetails
+
+	file, err := os.Open(pathToLockfile)
+	if err != nil {
+		return packages, fmt.Errorf("could not open %s: %w", pathToLockfile, err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+
+	var curPkg PackageDetails = PackageDetails{}
+
+	for scanner.Scan() {
```

Comment:
I'm wondering if we should take a similar approach to what I've done [in the `yarn.lock` parser](https://github.com/google/osv-scanner/blob/main/pkg/lockfile/parse-yarn-lock.go#L182-L196), which was to do the parsing in two phases: 1. group the lines for each package, 2. parse each collection of lines in `PackageDetails`.

While technically it's doing more loops, it should make it easier to reduce the amount of conditional branches (especially the duplicate ones outside the loop) which in my experience tends to be worth it - what do you think?

(if you do agree, then I think you should be able to use pretty much the exact same logic/structure as the `yarn.lock` parser, then add an extra check to handle if the name couldn't be found, which could go in a couple of different places)

Reply:
No problems for me, I'll try and get back to you soon.
I think that additional loops are not an issue since lockfiles are generally small.
There will be more code, but aligned to other parsers.

Reply:
Hello @G-Rath ,

I have a question:
 
> it should make it easier to reduce the amount of conditional branches
>(especially the duplicate ones outside the loop)

maybe I'm missing something but with your solution, aren't you performing the same pattern and number of conditional branches?
I'm referring to `parse-yarn-lock.go:` [here](https://github.com/google/osv-scanner/blob/main/pkg/lockfile/parse-yarn-lock.go#L32) and  [here](https://github.com/google/osv-scanner/blob/main/pkg/lockfile/parse-yarn-lock.go#L41)

Reply:
Hello @G-Rath ,
committed changes. Tests ok.
Thank you again for your advice and support.

Reply:
> maybe I'm missing something but with your solution, aren't you performing the same pattern and number of conditional branches?

Correct, but the footprint is smaller since it's just a length check + an append; the idea of that function is effectively to try and get us out of that raw unstructured state as fast as possible, so that we can do the more critical "business logic" stuff against a good structure which usually lets us avoid more complex versions of that pattern (which is where bugs can easily get introduced).

I view it as a lesser evil type situation - I won't say it's impossible to avoid having a duplicated check like that somewhere in this code, but I would expect it to have a significant tradeoff in at least one area (performance, readability, size, etc) that would make it worse than duplicating these three lines (I'm happy to be proven wrong though, if anyone has some ideas).